pkg/endpoint/provisioning.go

```diff
@@ -0,0 +1,41 @@
+package endpoint
+
+import "time"
+
+type ProvisioningRequest struct {
+	CertificateId *string
+	PickupId      *string
+	KeystoreId    *string
+	KeystoreName  *string
+	ProviderName  *string
+	Timeout       time.Duration
+}
+
+type ProvisioningMetadata interface {
+	GetAwsMetadata() AwsMetadata
+	GetAzureMetadata() AzureMetadata
+	GetGcpMetadata() GcpMetadata
+}
+
+type AwsMetadata interface {
+	GetARN() string
+}
+
+type AzureMetadata interface {
+	GetAKVID() string
+	GetCertificateName() string
+	GetVersion() string
+}
+
+type GcpMetadata interface {
+	GetGCPID() string
+	GetCertificateName() string
```

Comment:
So in this context we always assume this metadata is always for certificate metadata? I was adding `Certificate` to avoid confusion in case w e went further and added more info.

Reply:
Resolved by latest commit